### DISCUSSION TITLE
2008/06/04/%E4%BC%A0%E5%BC%95%E7%94%A8

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#222] [Announcements]
# 2008/06/04/%E4%BC%A0%E5%BC%95%E7%94%A8

传值是符合数据流驱动程序的传参方式，LabVIEW 中应该尽量使用这种方式。但是传引用在某些情况下是不可避免的，比如程序要在不同的线程中对同一数据进行操作，就得用到传引用。

https://blog.qizhen.xyz/2008/06/04/%E4%BC%A0%E5%BC%95%E7%94%A8

**Comment by ruanqizhen** [comment]
之前博客下的留言：

博弈
2011年09月7日 22:45 编辑
在用引用对数据进行改变时候总会出现意想不到的结果，有时候行不通，有时候开始运行时候能行，但关闭程序后再去运行，就有问题啦，是时间上还是什么回事，真还搞不懂，看您的东西，挺棒的，希望能找到一点答案，谢了您呢!!

回复
ruanqizhen
2011年09月15日 09:48 编辑
时有时无的错误，很可能是多线程引起的。LabVIEW自动多线程执行代码，代码执行顺序不确定，可能在某种顺序下会出错。
你试着用Sequence结构等方法把代码顺序固定下来看看能否解决问题。

回复
Pingback： 博客版《我和LabVIEW》目录 « 凡人琐忆 编辑
Qizhen
2009年04月9日 15:18 编辑
因为这是传引用的啊，所以不论在哪里运行子VI，它改动的都是同一块数据。所以并不需要把输出接入哪里。

回复
lihua
2009年04月8日 22:47 编辑
您好，关于队列的那个实例有一点没有弄明白，就是那个set number子VI，它有一个输入，输出引用，但是在主VI里面，它的输出并没有接入get data里面，但是程序运行后，结果却发生的改变，请问这是怎么回事，我是一个新手，请教啦！

回复
Qizhen
2009年03月11日 17:30 编辑
是的，在出栈时，会有复制。所以，如果不是必须传引用，可以直接传递大数组。只要子VI编写合理，可以避免数组数据拷贝。有些拷贝是不可避免的，比如取子数组等。

回复
gufei
2009年03月2日 11:28 编辑
请教大侠几个问题：1. 我想用队列保存一个大数组，想让他内存固定，对他进行处理，虽然是传引用了，但是每次要使用这个数组的还是要出栈，这样不还是肯定要复制了一下嘛？2.传入子VI是不是也是也要复制下数据、、、用labview处理大数据真的很头疼啊

回复
刘伟印
2008年08月22日 11:34 编辑
原来一直不太清楚Refnum是个什么东西，现在明白些了，谢谢！

回复
